src/songuploader/utils.py

```diff
@@ -17,25 +17,30 @@
 from uploader.models import Submission
 import os 
 
-from songuploader.settings import MEDIA_ROOT
-from uploader.models import Submission
-
-
 class ConfiguredLoginViewMixin(LoginRequiredMixin):
     login_url = reverse_lazy("login")
     redirect_field_name = "redirect_to"
 
 
 class LoginRequiredTemplateView(ConfiguredLoginViewMixin, TemplateView):
     template_name: str = None
-    
+
 def download_song(submission: Submission):
-    out_path = os.path.join("/tmp", f"{submission.user.username}.mp4")
-    YouTube(url=submission.song_url).streams.filter(only_audio=True).first().download(output_path=os.path.dirname(out_path), filename=os.path.basename(out_path))
-    open_file = open(out_path, 'rb')
+    mp4_out_path = os.path.join("/tmp", f"{submission.user.username}.mp4")
+    mp3_out_path = os.path.join("/tmp", f"{submission.user.username}.mp3")
+    YouTube(url=submission.song_url).streams.filter(only_audio=True).first().download(output_path=os.path.dirname(mp4_out_path), filename=os.path.basename(mp4_out_path))
+    os.system(f"ffmpeg -i {mp4_out_path} -vn {mp3_out_path}")
```

Comment:
This can create errors on unusual usernames e.g. containing dots or spaces

Reply:
Should i use the userid instead. Are there usernames where such behaviour is present

Reply:
👍